src/transaction/declare.rs

```diff
@@ -280,6 +297,33 @@ impl Declare {
             ),
         )
     }
+
+    pub(crate) fn create_for_simulation(
```

Comment:
looks like all the `create_for_simulation` methods doesn't need to take a reference to `self`. Since they take tx already